### PR TITLE
{k8s-extension} drop `azure-mgmt-relay` sdk dependency

### PR DESCRIPTION
**Pull request by ZengTaoxu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8s-extension

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py [hunk 1]
```diff
@@ -623,40 +624,49 @@ def _get_relay_connection_str(
...
         # create authorization rule
-        auth_rule_rights = [azure.mgmt.relay.models.AccessRights.manage,
-                            azure.mgmt.relay.models.AccessRights.send, azure.mgmt.relay.models.AccessRights.listen]
-        relay_client.hybrid_connections.create_or_update_authorization_rule(
-            resource_group_name, relay_namespace_name, hybrid_connection_name, auth_rule_name, rights=auth_rule_rights)
+        auth_rule_rights = ["Manage", "Send", "Listen"]
+        HycoAuthoCreate(cli_ctx=cmd.cli_ctx)(command_args={"resource_group": resource_group_name,
+                                                           "namespace_name": relay_namespace_name,
+                                                           "hybrid_connection_name": hybrid_connection_name,
+                                                           "name": hybrid_connection_name,
```
Comment (kairu-ms) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:661]: It should be `auth_rule_name` I think
```suggestion
                                                           "name": auth_rule_name,
```

**Review by kairu-ms — APPROVED** [review]

**Comment by yanzhudd** [comment]
Please rerun tests in live mode.

**Comment by ZengTaoxu** [comment]
> Please rerun tests in live mode.

These tests are in mode record_only.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**